### PR TITLE
passable matching functions 

### PR DESCRIPTION
**Pull request by Kelvinrr**

### CODE REVIEW

File: autocnet/matcher/subpixel.py [hunk 1]
```diff
@@ -313,8 +351,8 @@ def subpixel_transformed_template(sx, sy, dx, dy,
     template_size_x = template_size[0] * transform.scale[0]
     template_size_y = template_size[1] * transform.scale[1]
 
-    s_roi = roi.Roi(s_img, sx, sy, size_x=image_size[0], size_y=image_size[1])
-    d_roi = roi.Roi(d_img, dx, dy, size_x=template_size_x, size_y=template_size_y)
+    s_roi = roi.Roi(s_img, sx, sy, size_x=int(image_size[0]), size_y=int(image_size[1]))
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:354]: How are we getting non-integer image sizes here? Should we be handling the force to integer size in the check_image_size call on 348/349?

File: autocnet/matcher/subpixel.py [hunk 2]
```diff
@@ -974,6 +1020,10 @@ def geom_match_classic(base_cube,
     autocnet.matcher.subpixel.subpixel_phase: for list of kwargs that can be passed to the matcher
     """
 
+    print("in geommatch simple")
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1023]: remove?

File: autocnet/matcher/subpixel.py [hunk 3]
```diff
@@ -974,6 +1020,10 @@ def geom_match_classic(base_cube,
...
+    size_x = int(size_x)
+    size_y = int(size_y)
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1025]: Instead of casting, should we enforce `int` on the passed kwarg?

File: autocnet/matcher/subpixel.py [hunk 4]
```diff
@@ -998,8 +1048,7 @@ def geom_match_classic(base_cube,
         raise Exception(f"Window: {base_starty} < 0, center: {bcenter_x},{bcenter_y}")
 
     # specifically not putting this in a try/except, this should never fail
-    mlat, mlon = spatial.isis.image_to_ground(base_cube.file_name, bcenter_x, bcenter_y)
-    center_x, center_y = spatial.isis.ground_to_image(input_cube.file_name, mlon, mlat)[::-1]
+    center_x, center_y = bcenter_x, bcenter_y
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1051]: maybe a weird github rendering issue, but this looks like a syntax error.

File: autocnet/matcher/subpixel.py [hunk 5]
```diff
@@ -1017,16 +1066,14 @@ def geom_match_classic(base_cube,
                 return None, None, None, None, None
 
     base_gcps = np.array([*base_corners])
-    base_gcps[:,0] -= base_startx
-    base_gcps[:,1] -= base_starty
 
     dst_gcps = np.array([*dst_corners])
+    print(base_gcps, dst_gcps)
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1071]: remove? or is this logging important?

File: autocnet/matcher/subpixel.py [hunk 6]
```diff
@@ -1037,143 +1084,77 @@ def geom_match_classic(base_cube,
                     "Real" : "float64"
     }
 
-    base_pixels = list(map(int, [base_corners[0][0], base_corners[0][1], size_x*2, size_y*2]))
+    #base_pixels = list(map(int, [base_corners[0][0], base_corners[0][1], size_x*2, size_y*2]))
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1087]: remove comments as they are just old code?

File: autocnet/matcher/subpixel.py [hunk 7]
```diff
@@ -1037,143 +1084,77 @@ def geom_match_classic(base_cube,
...
-    dst_arr = input_cube.read_array(pixels=dst_pixels, dtype=dst_type)
+    dst_arr = input_cube.read_array(dtype=dst_type)
+
+    box = (0, 0, max(dst_arr.shape[1], base_arr.shape[1]), max(dst_arr.shape[0], base_arr.shape[0]))
+    dst_arr = np.array(Image.fromarray(dst_arr).crop(box))
 
     dst_arr = tf.warp(dst_arr, affine)
-    dst_arr = dst_arr[:size_y*2, :size_x*2]
+    print(base_arr.shape, dst_arr.shape )
+    print(bcenter_x, bcenter_y)
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1100]: same print question

File: autocnet/matcher/subpixel.py [hunk 8]
```diff
@@ -1436,11 +1417,12 @@ def subpixel_register_measure(measureid,
...
 def subpixel_register_point(pointid,
-                            subpixel_template_kwargs={},
                             cost_func=lambda x,y: 1/x**2 * y,
                             threshold=0.005,
                             ncg=None,
-                            version='new',
+                            geom_func='simple',
+                            match_func='classic',
+                            match_kwargs={},
                             **kwargs):
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1426]: I think we need to update the doc string here?

File: autocnet/matcher/subpixel.py [hunk 9]
```diff
@@ -1531,6 +1520,7 @@ def subpixel_register_point(pointid,
                     measure.ignore = True # Unable to geom match and no previous sucesses
                 continue
 
+            print(metric)
```
Comment (jlaura) [inline autocnet/matcher/subpixel.py:1523]: extra print? or we need text to describe what the random on screen float is.

File: autocnet/matcher/subpixel.py (hunk 1 above)
Comment (Kelvinrr) [reply autocnet/matcher/subpixel.py:354]: no idea? They seemed to be an issue before my previous PR that was causing a failure. I agree that is a better change.

File: autocnet/matcher/subpixel.py (hunk 4 above)
Comment (Kelvinrr) [reply autocnet/matcher/subpixel.py:1051]: been having weird merge problems recently, I don't recall this change having to be a part of this and the prints were also removed a while back...

File: autocnet/matcher/subpixel.py (hunk 1 above)
Comment (jlaura) [reply autocnet/matcher/subpixel.py]: :+1: Makes total sense.

**Comment by Kelvinrr** [comment]
@jlaura  you think this is good to merge?

**Review by jlaura — APPROVED** [review]

**Pull request merged**